src/SideNavigation/SideNav.js

```diff
@@ -57,11 +59,17 @@ SideNav.propTypes = {
     children: PropTypes.node,
     className: PropTypes.string,
     icons: PropTypes.bool,
-    selectedId: PropTypes.string
+    selectedId: PropTypes.string,
+    onItemSelect: PropTypes.func
+};
+
+SideNav.defaultProps = {
+    onItemSelect: () => { }
 };
 
 SideNav.propDescriptions = {
     icons: 'Set to **true** to only render icons for each `SideNavListItem`.',
+    onItemSelect: 'Prop callback to notify of internal state changes',
```

Comment:
Can we include something here that tells you what parameters are given (event and id)? - see HIG callbacks: https://hig.concur.com/pages/develop/web/components/nui-widgets/v11/popover.html#callbacks

Reply:
Agreed.  I would also make this a little more English-sounding (rather than purely technical).  Maybe something like "Callback function when a navigation item is selected. Arguments passed are the `event` and the `id` of the selected item."